File: android/app/src/main/java/com/gastosapp/data/local/ExpenseDao.java
package com.gastosapp.data.local;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;
import java.util.List;

@Dao
public interface ExpenseDao {
    @Insert
    void insert(ExpenseEntity expense);

    @androidx.room.Update
    void update(ExpenseEntity expense);

    @androidx.room.Delete
    void delete(ExpenseEntity expense);

    @Query("SELECT * FROM expenses WHERE id = :id")
    LiveData<ExpenseEntity> getExpenseById(int id);

    @Query("SELECT * FROM expenses ORDER BY timestamp DESC")
    LiveData<List<ExpenseEntity>> getAllExpenses();

    @Query("SELECT * FROM expenses ORDER BY timestamp DESC LIMIT 5")
    LiveData<List<ExpenseEntity>> getRecentExpenses();

    @Query("SELECT SUM(amount) FROM expenses WHERE timestamp >= :startTimestamp AND timestamp <= :endTimestamp")
    LiveData<Double> getTotalMonthlyExpenses(long startTimestamp, long endTimestamp);
}

File: app/src/main/kotlin/com/gastosapp/data/local/ExpenseDao.kt
package com.gastosapp.data.local

import androidx.lifecycle.LiveData
import androidx.room.*

@Dao
interface ExpenseDao {
    @Insert
    suspend fun insert(expense: ExpenseEntity)

    @Update
    suspend fun update(expense: ExpenseEntity)

    @Delete
    suspend fun delete(expense: ExpenseEntity)

    @Query("SELECT * FROM expenses WHERE id = :id")
    fun getExpenseById(id: Int): LiveData<ExpenseEntity>

    @Query("SELECT * FROM expenses ORDER BY timestamp DESC")
    fun getAllExpenses(): LiveData<List<ExpenseEntity>>

    @Query("SELECT * FROM expenses ORDER BY timestamp DESC LIMIT 5")
    fun getRecentExpenses(): LiveData<List<ExpenseEntity>>

    @Query("SELECT SUM(amount) FROM expenses WHERE timestamp >= :startTimestamp AND timestamp <= :endTimestamp")
    fun getTotalMonthlyExpenses(startTimestamp: Long, endTimestamp: Long): LiveData<Double>
}
